Python/Product/PythonTools/PythonTools/Repl/PythonCommonInteractiveEvaluator.cs

```diff
@@ -142,45 +142,45 @@ internal abstract class PythonCommonInteractiveEvaluator :
         }
 
 
-        public virtual VsProjectAnalyzer Analyzer {
-            get {
-                if (_analyzer != null) {
-                    return _analyzer;
-                }
+        public virtual VsProjectAnalyzer Analyzer => _analyzer;
 
-                var config = Configuration;
-                IPythonInterpreterFactory factory = null;
-                if (config?.Interpreter != null) {
-                    var interpreterService = _serviceProvider.GetComponentModel().GetService<IInterpreterRegistryService>();
-                    factory = interpreterService.FindInterpreter(config.Interpreter.Id);
-                }
+        public virtual async Task<VsProjectAnalyzer> GetAnalyzerAsync() {
+            if (_analyzer != null) {
+                return _analyzer;
+            }
 
-                if (factory == null) {
-                    _analyzer = _serviceProvider.GetPythonToolsService().DefaultAnalyzer;
-                } else {
-                    _analyzer = _serviceProvider.GetUIThread().InvokeTaskSync(async () => {
-                        var pyProject = GetAssociatedPythonProject(config.Interpreter);
-
-                        var a = await VsProjectAnalyzer.CreateForInteractiveAsync(
-                            _serviceProvider.GetComponentModel().GetService<PythonEditorServices>(),
-                            factory,
-                            DisplayName.IfNullOrEmpty("Unnamed")
-                        );
-
-                        IEnumerable<string> sp;
-                        if (pyProject != null) {
-                            sp = pyProject.GetSearchPaths();
-                        } else {
-                            var sln = _serviceProvider.GetService(typeof(SVsSolution)) as IVsSolution;
-                            sp = sln?.EnumerateLoadedPythonProjects().SelectMany(p => p.GetSearchPaths()).ToArray();
-                        }
-                        await a.SetSearchPathsAsync(sp.MaybeEnumerate());
+            var config = Configuration;
+            IPythonInterpreterFactory factory = null;
+            if (config?.Interpreter != null) {
+                var interpreterService = _serviceProvider.GetComponentModel().GetService<IInterpreterRegistryService>();
+                factory = interpreterService.FindInterpreter(config.Interpreter.Id);
+            }
 
-                        return a;
-                    }, CancellationToken.None);
-                }
-                return _analyzer;
+            if (factory == null) {
+                _analyzer = await _serviceProvider.GetPythonToolsService().GetSharedAnalyzerAsync();
+            } else {
+                _analyzer = await _serviceProvider.GetUIThread().InvokeTask(async () => {
```

Comment:
Is it possible that this code is called twice before `_analyzer` is assigned?

Reply:
Right now this function is not called... guess I should fix that first :)